src/main/scala/scorex/core/network/DeliveryTracker.scala

```diff
@@ -12,89 +14,178 @@ import scala.util.{Failure, Try}
 
 
 /**
-  * This class tracks modifier ids which are expected from and delivered by other peers
-  * in order to ban or de-prioritize peers which are delivering what is not expected
+  * This class tracks modifier statuses.
+  * Modifier can be in one of the following states: Unknown, Requested, Received, Applied, Invalid. See
+  * ModifiersStatus for states description.
+  * Modifiers in `Requested` state are kept in `requested` map containing info about peer and number of retries.
+  * Modifiers in `Received` state are kept in `received` set.
+  * Modifiers in `Invalid` state are kept in `invalid` set to prevent this modifier download and processing.
+  * Modifiers in `Applied` state are not kept in this class - we can get this status from object, that contains
```

Comment:
Based on the description, it is not logical to call the status "Applied". Better to use something like "External".